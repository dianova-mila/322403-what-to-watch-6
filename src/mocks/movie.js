export default {
  name: `Macbeth`,
  posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Macbeth.jpg`,
  previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/macbeth.jpg`,
  backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Macbeth.jpg`,
  backgroundColor: `#F1E9CE`,
  description: `Macbeth, the Thane of Glamis, receives a prophecy from a trio of witches that one day he will become King of Scotland. Consumed by ambition and spurred to action by his wife, Macbeth murders his king and takes the throne for himself.`,
  rating: 3.3,
  scoresCount: 48798,
  director: `Justin Kurzel`,
  starring: [
    `Michael Fassbender`,
    `Marion Cotillard`,
    `Jack Madigan`
  ],
  runTime: 113,
  genre: `Drama`,
  released: 2015,
  id: `1`,
  isFavorite: false,
  videoLink: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
  previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
};
