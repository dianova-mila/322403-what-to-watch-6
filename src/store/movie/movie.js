import {createReducer} from "@reduxjs/toolkit";
import {loadOneMovie, loadPromoMovie, loadComments} from "../actions";

const initialState = {
  movie: {},
  promoMovie: {},
  isPromoMovieLoaded: false,
  comments: [],
  isOneMovieLoaded: false,
};

const movie = createReducer(initialState, (builder) => {
  builder.addCase(loadOneMovie, (state, action) => {
    state.movie = action.payload;
    state.isOneMovieLoaded = true;
  });

  builder.addCase(loadPromoMovie, (state, action) => {
    state.promoMovie = action.payload;
    state.isPromoMovieLoaded = true;
  });

  builder.addCase(loadComments, (state, action) => {
    state.comments = action.payload;
  });
});

export {movie};
