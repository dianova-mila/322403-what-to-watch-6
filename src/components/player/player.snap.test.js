import React from "react";
import {Router} from "react-router-dom";
import {render} from "@testing-library/react";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import {createMemoryHistory} from "history";
import movie from "../../mocks/movie";
import App from "../app/app";

const mockStore = configureStore({});


it(`Player should render correctly`, () => {
  const store = mockStore({
    MOVIE: {movie, isOneMovieLoaded: true}
  });

  const history = createMemoryHistory();
  history.push(`/player/1`);

  const {container} = render(
      <Provider store={store}>
        <Router history={history}>
          <App />
        </Router>
      </Provider>
  );

  expect(container).toMatchSnapshot();
});
