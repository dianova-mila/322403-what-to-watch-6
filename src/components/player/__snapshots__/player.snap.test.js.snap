// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player should render correctly 1`] = `
<div>
  <div
    class="player"
  >
    <video
      autoplay=""
      class="player__video"
      poster="https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Macbeth.jpg"
      src="http://media.xiph.org/mango/tears_of_steel_1080p.webm"
    />
    <button
      class="player__exit"
      type="button"
    >
      Exit
    </button>
    <div
      class="player__controls"
    >
      <div
        class="player__controls-row"
      >
        <div
          class="player__time"
        >
          <progress
            class="player__progress"
            max="100"
            value="0"
          />
          <div
            class="player__toggler"
            style="left: 0%;"
          >
            Toggler
          </div>
        </div>
        <div
          class="player__time-value"
        >
          0:00:00
        </div>
      </div>
      <div
        class="player__controls-row"
      >
        <button
          class="player__play"
          type="button"
        >
          <svg
            height="19"
            viewBox="0 0 19 19"
            width="19"
          >
            <use
              xlink:href="#pause"
            />
          </svg>
          <span>
            Play
          </span>
        </button>
        <div
          class="player__name"
        >
          Macbeth
        </div>
        <button
          class="player__full-screen"
          type="button"
        >
          <svg
            height="27"
            viewBox="0 0 27 27"
            width="27"
          >
            <use
              xlink:href="#full-screen"
            />
          </svg>
          <span>
            Full screen
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
